packages/pigly/src/_provider.ts

```diff
@@ -6,5 +6,5 @@ export interface IProvider<T> {
 
 export type ProviderWrap<T> = T extends any[]
   ? {
-    [P in keyof T]: IProvider<T[P]>;
+    [P in keyof T]?: IProvider<T[P]>;
```

Comment:
shouldn't be optional - so that `toClass` errors if you use it without proper provider signatures